src/EventSubscriber/ExceptionSubscriber.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Drupal\elastic_apm\EventSubscriber;
+
+use Drupal\elastic_apm\ApiServiceInterface;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * The ElasticApm exception event subscriber class.
+ *
+ * @package Drupal\elastic_apm\EventSubscriber
+ */
+class ExceptionSubscriber implements EventSubscriberInterface {
+
+  /**
+   * The Elastic APM service object.
+   *
+   * @var \Drupal\elastic_apm\ApiServiceInterface
+   */
+  protected $apiService;
+
+  /**
+   * The actual PHP Agent.
+   *
+   * @var \PhilKra\Agent
+   */
+  protected $phpAgent;
+
+  /**
+   * Constructs a ExceptionSubscriber object.
+   *
+   * @param \Drupal\elastic_apm\ApiServiceInterface $api_service
+   *   The Elastic APM service object.
+   */
+  public function __construct(ApiServiceInterface $api_service) {
+    $this->apiService = $api_service;
+
+    // Initialize the PHP agent if the Elastic APM config is configured.
+    if ($this->apiService->isEnabled() && $this->apiService->isConfigured()) {
+      $this->phpAgent = $this->apiService->getAgent();
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return [
+      KernelEvents::EXCEPTION => ['onException', -300],
+    ];
+  }
+
+  /**
+   * Send the exception to the PHP Agent whenever an exception is triggered.
+   *
+   * @param \Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent $event
+   *   The request event object.
+   */
+  public function onException(GetResponseForExceptionEvent $event) {
+    // Return if Elastic isn't enabled.
+    if (!$this->apiService->isEnabled()) {
+      return;
+    }
+
+    // Don't process if Elastic APM is not configured.
+    if (!$this->apiService->isConfigured()) {
+      return;
+    }
+
+    // Only log the exception if the captureExceptions config is checked.
+    if (!$this->apiService->getConfig()['captureExceptions']) {
+      return;
+    }
```

Comment:
**CHANGE REQUEST**

It's bad API design to provide public access to the full configuration. We should provide methods that allows callers to get the information they need. Similar to `isEnabled/isConfigured` we should have a function `captureExceptions` or `captureThrowable` method that we can call here. I'd suggest to either keep the configuration setting as is and name the function `captureExceptions()` or follow the agent's names and rename the configuration setting AND the new method to `captureThrowable()`.